lib/services/webpack/webpack-compiler-service.ts

```diff
@@ -260,7 +260,10 @@ export class WebpackCompilerService extends EventEmitter implements IWebpackComp
 		// if files will be emitted or not. This way, the first successful compilation after fixing the compilation error generates
 		// a hash that is not the same as the one expected in the latest emitted hot-update.json file.
 		// As a result, the hmr chain is broken and the changes are not applied.
-		const isHashValid = nextHash ? this.expectedHashes[platform] === currentHash : true;
+		let isHashValid = nextHash ? this.expectedHashes[platform] === currentHash : true;
+		if (this.expectedHashes[platform] === nextHash) {
+			isHashValid = true;
```

Comment:
IMO, this is not a valid hash, we've already handled it. I belive this logic should be directly in the watch callback and if `currentHash` and `nextHash` are equal, we should not call `getUpdatedEmittedFiles` and `emmit` anything at all.